app/jenkins-for-jira-ui/src/components/ConnectJenkins/ConnectJenkins/ConnectJenkins.tsx

```diff
@@ -35,9 +39,15 @@ const ConnectJenkins = () => {
 	}, [uuid]);
 
 	useEffect(() => {
+		const analyticsClient = new AnalyticsClient();
```

Comment:
Better to define `analyticsClient` outside of useEffect, as it just needs to be defined once.

Reply:
You can't do that with useEffect. The object construction of analyticsClient makes the dependencies in the useEffect arrat change on every render so it has to be defined like this so it doesn't need to be included in the array and cause side effects.